identity-card-credential-displayer.js

```diff
@@ -0,0 +1,270 @@
+/*!
+ * Identity Card Offer Displayer
+ *
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+/* global requirejs */
+define(['angular', 'jsonld', 'JsBarcode'], function(angular, jsonld, JsBarcode) {
+
+'use strict';
+
+function register(module) {
+  module.component('brIdentityCardCredentialDisplayer', {
+    bindings: {
+      model: '<brModel',
+      library: '<?brLibrary',
+      options: '<brOptions'
+    },
+    controller: Ctrl,
+    templateUrl:
+      requirejs.toUrl(
+        'bedrock-angular-card-displayer/identity-card-credential-displayer.html')
+  });
+}
+
+var DISPLAY_CONTEXT = [
+  'https://w3id.org/identity/v1',
+  'https://w3id.org/credentials/v1',
+  'https://w3id.org/documents/v1'
+];
+
+var DISPLAY_FRAME = {
+  '@context': DISPLAY_CONTEXT,
+  claim: {}
+};
+/* @ngInject */
+function Ctrl($scope, $filter, brCardDisplayerService) {
+  var self = this;
+
+  self.cardStyle = {
+    width: '300px'
+  };
+
+  self.pictureStyle = {
+    position: 'absolute',
+    right: '15px',
+    top: '60px',
+    height: '75px'
+  };
+
+  self.barcodeStyle = {
+    position: 'absolute',
+    left: '18px',
+    top: '128px',
+    height: '30px',
+    width: '150px'
+  };
+
+  self.jobTitleStyle = {
+    position: 'absolute',
+    top: '40px',
+    right: '22px',
+    fontSize: '5em'
+  };
+
+  self.dateStyle = {
+    position: 'absolute',
+    top: '140px',
+    right: '26px',
+    fontSize: '4.5em'
+  };
+
+  self.idNumberStyle = {
+    position: 'absolute',
+    top: '110px',
+    left: '18px',
+    fontSize: '5em'
+  };
+
+  self.bottomNameStyle = {
+    position: 'absolute',
+    top: '96px',
+    left: '18px',
+    fontSize: '5em'
+  };
+
+  self.topNameStyle = {
+    position: 'absolute',
+    top: '82px',
+    left: '18px',
+    fontSize: '5em'
+  };
+
+  self.containerStyle = {};
+
+  self.containerClass = {
+    'br-card-id-1-border': true,
+    'br-card-id-1-radial-gradient': true
+  };
+
+  self.textStyle = {};
+  self.documentType = {};
+  self.documentType.name = '';
+  // internal vars for tracking style changes
+  var style = {};
+  var styleOptions = {
+    background: {
+      radialGradient: false
+    }
+  };
+
+  self.$onChanges = function(changes) {
+    if(changes.model && changes.model.currentValue) {
+      jsonld.promises.frame(
```

Comment:
The current pattern is to wrap things that return a `Promise` in a `$q` style promise by wrapping with `$q.resolve(...)`

When this is done, there is no need to add `$scope.$apply()` anywhere in the promise chain.